pkg/broker/broker.go

```diff
@@ -339,12 +339,38 @@ func (a AnsibleBroker) Bind(instanceUUID uuid.UUID, bindingUUID uuid.UUID, req *
 		Parameters: &params,
 	}
 
+	// Verify we're not rebinding the same instance. if err is nil, there is an
+	// instance. Let's compare it to the instance we're being asked to bind.
+	//
+	// if err is not nil, we will just bubble that up
+	//
 	// if binding instance exists, and the parameters are the same return: 200.
 	// if binding instance exists, and the parameters are different return: 409.
 	//
 	// return 201 when we're done.
-	//
-	// once we create the binding instance, we call apb.Bind
+	if bi, err := a.dao.GetBindInstance(bindingUUID.String()); err == nil {
+		if bi.Id.String() == bindingInstance.Id.String() &&
+			reflect.DeepEqual(bi.Parameters, bindingInstance.Parameters) {
+
+			a.log.Debug("already have this binding instance returning 200")
+			return &BindResponse{}, ErrorAlreadyProvisioned
+		} else if bi.Id.String() == bindingInstance.Id.String() &&
+			!reflect.DeepEqual(bi.Parameters, bindingInstance.Parameters) {
+
+			// TODO: remove these debug statements at some point
+			a.log.Debug("Existing binding parameters")
```

Comment:
If you aren't using these  for something else, maybe we should drop them so we don't log any bind params.  We should keep that info secret between the broker and apb.